File: python/array_problem/jump_game.py
def can_jump(nums):

    steps = 0
    length = len(nums) - 1

    for k, v in enumerate(nums):

        if k > steps: 
            return False

        steps = max(steps, k + v)

    # After the loop, check if the farthest reachable index is at or beyond the last index
    return steps >= length




def get_min_jump(list):

    steps = 0
    maxJamp = 0
    currentJumpEndAt = 0
    length = len(list)


    for k,v in enumerate(list):

        maxJamp = max(maxJamp, (k + v ))

        if currentJumpEndAt == k:

            currentJumpEndAt = maxJamp
            
            steps +=1

        if currentJumpEndAt >= (length -1):

            return steps;

File: js/src/ArrayProblem/JumpGame.js




export function canJumpGame(list){

	let steps = 0;
	let length = list.length - 1;

	for (let i = 0; i <= length; i++) {
			
			if (i > steps) {
				return false;
			}

			steps = Math.max(steps, (i + list[i]));

			if (steps >= length) {return true}
	}
	return false;
}


export function getMinJump($list){

	let steps = 0;
	let maxJump = 0;
	let length = $list.length;
	let currentJumpEndAt = 0;

	for (let i = 0;i < length; i++) {
		
		maxJump = Math.max(maxJump,(i+$list[i]));

		if (currentJumpEndAt === i) {
			steps++;
			currentJumpEndAt = maxJump;
		}

		if (currentJumpEndAt >= (length -1)) return steps;
	}
}